python/ray/_private/serialization.py

```diff
@@ -233,16 +234,19 @@ def _deserialize_actor_died_error(self, data, metadata_fields):
     def _deserialize_object(self, data, metadata, object_ref):
         if metadata:
             metadata_fields = metadata.split(b",")
+
+            # Hock for new serialization infrastructure, will move the original code
+            # to here step by step.
+            if metadata_fields[0] == ray_constants.OBJECT_METADATA_TYPE_NEW_PROTOCOL:
+                result = RaySerializationResult.from_bytes(data.to_pybytes())
+                return serialization_new._deserialize(result)
+
             if metadata_fields[0] in [
                 ray_constants.OBJECT_METADATA_TYPE_CROSS_LANGUAGE,
                 ray_constants.OBJECT_METADATA_TYPE_PYTHON,
             ]:
```

Comment:
How hard is it to port these two old serializers to the new infra? Seems to me that's the proof in the pudding needed: whether we can pass tests with the old code removed.